e2e-tests/gef/cypress/e2e/facility/facility-end-date.spec.js

```diff
@@ -0,0 +1,250 @@
+import relative from '../relativeURL';
+import facilityEndDate from '../pages/facility-end-date';
+import aboutFacility from '../pages/about-facility';
+import {
+  tomorrowDay,
+  tomorrowMonth,
+  tomorrowYear,
+  todayDay,
+  todayMonth,
+  todayYear,
+  yesterdayDay,
+  yesterdayMonth,
+  yesterdayYear,
+} from '../../../../e2e-fixtures/dateConstants';
+import { BANK1_MAKER1 } from '../../../../e2e-fixtures/portal-users.fixture';
+
+const applications = [];
+let token;
+
+const now = new Date();
+const nextYear = Number(todayYear) + 1;
+
+const facilityEndDateEnabled = Number(Cypress.env('GEF_DEAL_VERSION')) >= 1;
+
+context('Facility End Date Page', () => {
+  let application;
+  let facilityId;
+
+  before(() => {
+    cy.loadData();
+    cy.apiLogin(BANK1_MAKER1)
+      .then((tok) => {
+        token = tok;
+      })
+      .then(() => cy.apiFetchAllGefApplications(token))
+      .then(({ body }) => {
+        body.items.forEach((item) => {
+          cy.apiFetchAllFacilities(item._id, token).then((res) => {
+            applications.push({
+              id: item._id,
+              facilities: res.body.items.filter((it) => it.details.dealId === item._id),
+            });
+          });
+        });
+      });
+    cy.login(BANK1_MAKER1);
+  });
+
+  beforeEach(() => {
+    application = applications[1];
+    facilityId = application.facilities[0].details._id;
+
+    cy.saveSession();
+  });
+
+  if (facilityEndDateEnabled) {
+    it('displays the correct elements', () => {
+      cy.visit(relative(`/gef/application-details/${application.id}/facilities/${facilityId}/about-facility`));
+
+      aboutFacility.isUsingFacilityEndDateYes().click();
+      aboutFacility.saveAndReturnButton().click();
+
+      cy.visit(relative(`/gef/application-details/${application.id}/facilities/${facilityId}/facility-end-date`));
+
+      facilityEndDate.backLink();
+      facilityEndDate.headingCaption();
+      facilityEndDate.facilityEndDateDay();
+      facilityEndDate.facilityEndDateMonth();
+      facilityEndDate.facilityEndDateYear();
+      facilityEndDate.facilityEndDateDetails();
+      facilityEndDate.continueButton();
+      facilityEndDate.saveAndReturnButton();
+    });
+
+    it('redirects the user to `About this facility` page when clicking on back link', () => {
+      cy.visit(relative(`/gef/application-details/${application.id}/facilities/${facilityId}/facility-end-date`));
+
+      facilityEndDate.backLink().click();
+
+      cy.url().should('eq', relative(`/gef/application-details/${application.id}/facilities/${facilityId}/about-facility`));
+    });
+
+    it('validates form when clicking on Continue', () => {
```

Comment:
`validates the form` sorry -- pet peeve here, I dislike unneeded shortening as it saves 2 seconds but adds a very slight cognative load for all maintainers...